### PR TITLE
fix(cloudformation): fix some resources

### PR DESCRIPTION
**Pull request by nikpivkin**
Fix https://github.com/aquasecurity/trivy/issues/5802

CloudFormation casts the types to the required types if possible. So a boolean property can be passed the string `"false"`, which will be cast to bool. We could infer the type at the parsing stage so that we don't have to worry about it later, but we don't know the type of the property and so it's not safe. We must use `GetBoolProperty` to get the boolean value of the property, because this method converts the string representation of a boolean value to a boolean value.

Also fixed incorrect property names and default property values.

### CODE REVIEW

File: internal/adapters/cloudformation/aws/ec2/adatp_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package ec2
```
Comment (simar7) [inline internal/adapters/cloudformation/aws/ec2/adatp_test.go:1]: nit: typo in file name, should be `adapt_test.go`

File: internal/adapters/cloudformation/aws/rds/cluster.go [hunk 2]
```diff
@@ -9,56 +9,26 @@ import (
...
-			BackupRetentionPeriodDays: defsecTypes.IntDefault(1, clusterResource.Metadata()),
-			ReplicationSourceARN:      defsecTypes.StringDefault("", clusterResource.Metadata()),
+			BackupRetentionPeriodDays: clusterResource.GetIntProperty("BackupRetentionPeriod", 1),
+			ReplicationSourceARN:      clusterResource.GetStringProperty("SourceDBInstanceIdentifier"),
 			PerformanceInsights: rds.PerformanceInsights{
 				Metadata: clusterResource.Metadata(),
-				Enabled:  defsecTypes.BoolDefault(false, clusterResource.Metadata()),
-				KMSKeyID: defsecTypes.StringDefault("", clusterResource.Metadata()),
+				Enabled:  clusterResource.GetBoolProperty("PerformanceInsightsEnabled"),
+				KMSKeyID: clusterResource.GetStringProperty("PerformanceInsightsKmsKeyId"),
```
Comment (simar7) [inline internal/adapters/cloudformation/aws/rds/cluster.go:19]: should this be the following?

```suggestion
				KMSKeyID: clusterResource.GetStringProperty("PerformanceInsightsKMSKeyId"),
```

Comment (simar7) [reply internal/adapters/cloudformation/aws/rds/cluster.go:19]: Either way, we should add it to the test case so we can cover it.

Comment (nikpivkin) [reply internal/adapters/cloudformation/aws/rds/cluster.go:19]: Must be [PerformanceInsightsKmsKeyId](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-rds-dbcluster.html#cfn-rds-dbcluster-performanceinsightskmskeyid)

Comment (nikpivkin) [reply internal/adapters/cloudformation/aws/rds/cluster.go:18]: [Done](https://github.com/aquasecurity/trivy-iac/pull/69/commits/f7d88abf8367ee594a48ebce787a120eaea34017#diff-613168e6eaf30167b7831f0744245b71bbcf03248176424b6be56bb3eb2d4282)

File: internal/adapters/cloudformation/aws/ec2/adatp_test.go (hunk 1 above)
Comment (nikpivkin) [reply internal/adapters/cloudformation/aws/ec2/adatp_test.go]: fixed https://github.com/aquasecurity/trivy-iac/pull/69/commits/20a2496e11dd6e3e6b958b12c8953a439432b72a

**Review by simar7 — APPROVED** [review]

**Pull request merged**